http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HttpResponseListener.java

```diff
@@ -61,15 +60,8 @@ public void onMessage(HTTPCarbonMessage httpResponseMessage) {
                     .createHttpResponse(httpResponseMessage, connectionCloseAfterResponse);
             sourceContext.write(response);
 
-            while (true) {
-                if (httpResponseMessage.isEndOfMsgAdded() && httpResponseMessage.isEmpty()) {
-                    ChannelFuture future = sourceContext.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
-                    if (connectionCloseAfterResponse) {
-                        future.addListener(ChannelFutureListener.CLOSE);
-                    }
-                    break;
-                }
-                HttpContent httpContent = httpResponseMessage.getHttpContent();
+            httpResponseMessage.getHttpContentAsync().setMessageListener(httpContent ->
+                    this.sourceContext.channel().eventLoop().execute(() -> {
                 if (httpContent instanceof LastHttpContent) {
```

Comment:
Move this logic to a separate private function, such as "isLastHTTPContent". This same logic has been used in multiple places, and a single common place is useful to make sure it's consistent everywhere.